include/type.hpp

```diff
@@ -188,6 +188,12 @@ class RecordType : public Type {
   /// @return The type id.
   virtual std::string id()  // NOLINT(readability-identifier-naming)
       const noexcept = 0;
+  /// @return The type offset in the record based on `id`.
```

Comment:
Please do state that this function throws under what kind of situations, and what type of exception it throws. For example,
```suggestion
  /// @return The type offset in the record based on `id`.
  /// @throw `std::runtime_error` if the `id` is not a member of the record.
```